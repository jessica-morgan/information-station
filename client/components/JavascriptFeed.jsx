import React, { useState, useEffect } from 'react'
import { getJavascriptNewsFeed } from '../api/newsApi'

// this component should display a list of js (just the title)
// create seperate componet for individual articles
const JavascriptFeed = () => {
  const [jsFeed, setJsFeed] = useState()

  useEffect(() => {
    getJavascriptNewsFeed()
      .then(articles => {
        setJsFeed({
          javascriptNews: articles
        })
      })
  }, [])
  return jsFeed ? (
    <div style={{ display: 'block', width: '70vw', height: '100%', float: 'right', position: 'relative' }}>
      {jsFeed.javascriptNews.map((article, idx) => {
        return (
          <div key={idx}>
            <h1 key={article.title}>{article.title}</h1>
          </div>
        )
      })}
    </div>
  )
    : <div style={{ display: 'block', width: '70vw', height: '100%', float: 'right', position: 'relative' }}>
         loading component
    </div>
}

export default JavascriptFeed
