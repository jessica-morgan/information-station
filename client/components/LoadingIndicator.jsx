import React from 'react'

const LoadingIndicator = () => {
  return (
    <div>
      <img src="/loading.gif"></img>
    </div>
  )
}

export default LoadingIndicator
