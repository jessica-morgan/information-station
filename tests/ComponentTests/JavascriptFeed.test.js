import React from 'reactn'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import JavascriptFeed from '../../client/components/JavascriptFeed'
import { mount } from 'enzyme'
import LoadingIndicator from '../../client/components/LoadingIndicator'

test('IsItBullshit renders correctly', () => {
  const component = renderer.create(
    <MemoryRouter>
      <JavascriptFeed />
    </MemoryRouter>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('IsItBullshit should render LoadingIndicator if posts is not yet set with api data', done => {
  const wrapper = mount(<JavascriptFeed />)
  setImmediate(() => {
    wrapper.update()
    expect(wrapper.containsMatchingElement(<LoadingIndicator></LoadingIndicator>)).toBeTruthy()
    done()
  })
})

it('h1 should call selectTitle and currentTitleSelected onClick', () => {
  const selectTitleMock = jest.fn()
  const currentTitleSelectedMock = jest.fn()
  const wrapper = mount(<h1 onClick={() => { selectTitleMock(); currentTitleSelectedMock() }}/>)
  wrapper.find('h1').simulate('click')
  expect(selectTitleMock).toBeCalled()
  expect(currentTitleSelectedMock).toBeCalled()
})