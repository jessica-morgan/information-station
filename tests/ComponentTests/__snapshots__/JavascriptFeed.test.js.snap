// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JavascriptRedditFeed feed renders correctly 1`] = `
<div>
  <div>
    <img
      className="loading-indicator"
      src="/loading.gif"
    />
  </div>
</div>
`;
